### PR TITLE
Add FillRectangle function to display driver

### PR DESCRIPTION
**Pull request by lsd-techno**
## Description

- Add FillRectangle function
- Update Write function to use FillRectangle function instead of slow and little bit 'buggy' DrawRoundedRectangle function
- Above fixes damaged background for characters printed with 'Display:Write' function in most cases

## Motivation and Context

- Wish to have better and faster display updates.

- Future development based on #2413.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Comment by josesimoes** [comment]
@lsd-techno a quick tutorial on how to reference GitHub issues and PRs on comments 😉 
1) if it's from the same repo: just used `#NNNN` ->>>>> #2413 
2) from a different repo need to point to it, like this `nanoframework/nf-interpreter#2413` ->>> nanoframework/nf-interpreter#2413 

GitHub takes care of doing the formating, don't need to worry about that.
(and make sure to use our template and just fill in the blanks, everything will come up just OK)

File: src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ HRESULT Library_nanoFramework_Graphics_nanoFramework_UI_DisplayControl::
 {
     NANOCLR_HEADER();
     CLR_INT32 orientation = stack.Arg0().NumericByRef().s4;
-    g_GraphicsDriver.ChangeOrientation((DisplayOrientation)orientation);
+    stack.SetResult_Boolean(g_GraphicsDriver.ChangeOrientation((DisplayOrientation)orientation));
```
Comment (josesimoes) [inline src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp:67]: This is probably fixing a bug with the `SetResult_Boolean` not being called.

Please make this a separate PR because its not related with the FillRectangle that this PR is about.

File: src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp [hunk 2]
```diff
@@ -213,11 +213,11 @@ HRESULT Library_nanoFramework_Graphics_nanoFramework_UI_DisplayControl::
         backgroundColor = stack.Arg7().NumericByRef().u4;
 
         // Prepare all the elements for the colors
-        GFX_Pen pen;
+        // GFX_Pen pen;
```
Comment (josesimoes) [inline src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp:216]: All the code modifications here seem to be changing the `Write___STATIC__VOID__STRING__U2__U2__U2__U2__nanoFrameworkUIFont__nanoFrameworkPresentationMediaColor__nanoFrameworkPresentationMediaColor`  behavior. 🤔 

This function is used in generic UI DisplayControl. A couple of thoughts:
- If this is fixing/improving something on the generic UI behavior then it belongs to a separate PR.
- If it's required specifically for the fix on FilRectangle, then it needs to be validated with other displays as well.
- ANyways, if this commented out code as a purpose or needs to be worked on, please leave a comment with a clear explanation, otherwise just remove it.

File: src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp (hunk 1 above)
Comment (lsd-techno) [reply src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp:67]: This PR is based is based on previous PR #2413 where `SetResult_Boolean` fixed already.

File: src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp (hunk 2 above)
Comment (lsd-techno) [reply src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp]: Yes. `FillRectangle` function add to be used in `Write` function only for the moment.
Cleaned up.

**Comment by lsd-techno** [comment]
> @lsd-techno a quick tutorial on how to reference GitHub issues and PRs on comments 😉
> 
> 1. if it's from the same repo: just used `#NNNN` ->>>>> [Fix display driver switch orientation #2413](https://github.com/nanoframework/nf-interpreter/pull/2413)
> 2. from a different repo need to point to it, like this `nanoframework/nf-interpreter#2413` ->>> [Fix display driver switch orientation #2413](https://github.com/nanoframework/nf-interpreter/pull/2413)
> 
> GitHub takes care of doing the formating, don't need to worry about that. (and make sure to use our template and just fill in the blanks, everything will come up just OK)

Thanks for the tips.
Next times will not make changes in PR template too.

**Review by Ellerbach — APPROVED** [review]
Looks good to me! Thanks a lot!

File: src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp (hunk 1 above)
Comment (josesimoes) [reply src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp:67]: Understood. But if it's there, shouldn't be here... 
Ideally we want PRs to be:
- _truthful_  to the PR title and not adding _just_ a couple of tiny other fixes, just because one has stumbled on those
- atomic: a PR will only change things that are meant/required

This may sound picky, but it's not! 😅 In case a PR needs to be reverted by some reason, if it's touching unrelated stuff it may end up causing regressions, force extra work on picking up those tiny other changes to be placed back in and other unforeseen changes/issues.

Comment (lsd-techno) [reply src/nanoFramework.Graphics/Graphics/Native/nanoFramework_Graphics_nanoFramework_UI_DisplayControl.cpp]: Removed.

**Review by josesimoes — APPROVED** [review]
LGTM!
